mateusfreira/nun-db
# Implement join address param

## Add join address command
- [x] When joining as a replica the server sends its tcp address to the primary
- [x] Add new command line option
- [x] Replace the command line to use the new parameter if it is set
* Defines the command line parameter
[src/lib/commad_line/commands.rs:59 ]
Used here to pass to start db
[src/bin/main.rs:58]
Replicate join here
[src/lib/replication_ops.rs:501 ]
Change here to use the new parameter
[src/bin/main.rs:117]
Ask to join
[src/lib/replication_ops.rs:745 ]
[src/bin/main.rs:135 ] Called at main
- [x] How to test this??? Write integration tests
- [x] Test it in the docker
- [x] ack 1697328068431833087 nun-db-secondary-1:3018 must use the join addr not the tcp address



## Open problems
In the docker setup it is not passing via the proxy to send data between the nodes



## To test

```
docker-compose up
```

```
make connect-primary
Connecting to primary
auth mateus mateus
valid auth
ok
metrics-state
metrics-state pending_ops: 4, op_log_file_size: 150, op_log_count: 6,replication_time_moving_avg: 0.0, get_query_time_moving_avg: 0.956429945940613

ok
```
replication_time_moving_avg should be close to one
